File: chat-ecommerce-ui/src/app/models/chat.models.ts
export interface Product {
  id: number;
  sku?: string;
  productName: string;
  description?: string;
  category?: string;
  subcategory?: string;
  brand?: string;
  tags?: string;
  currencyCode?: string;
  imageUrl?: string;
  price: number;
  availableStock: number;
  relevanceScore?: number;
}

export interface ChatMessageModel {
  content: string;
  role: 'user' | 'assistant';
}

export interface ChatResponseMessage extends ChatMessageModel {
  type?: 'text' | 'products';
  data?: Product[];
  suggestedReplies?: string[];
  clarifyingQuestion?: string;
  nextAction?: string;
  sessionId?: string;
}

export interface ChatRequestModel extends ChatMessageModel {

}

export interface QuizQuestionOption {
  key: string;
  label: string;
}

export interface QuizQuestion {
  id: string;
  prompt: string;
  options: QuizQuestionOption[];
}

export interface QuizStartResponse {
  sessionId: string;
  quizType: string;
  question: QuizQuestion;
}

export interface QuizResult {
  quizType: string;
  personaKey: string;
  personaLabel: string;
  confidence: number;
  recommendedTags: string[];
}

export interface QuizAnswerResponse {
  completed: boolean;
  nextQuestion?: QuizQuestion;
  result?: QuizResult;
  recommendedProducts?: Product[];
}

File: ChatBotService/Models/ChatModels.cs
namespace ChatBotService.Models;

public class ProductInfo
{
    public int Id { get; set; }
    public string ProductName { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int AvailableStock { get; set; }
    public double RelevanceScore { get; set; }
}

public class ChatResponseModel
{
    public string Message { get; set; } = string.Empty;
    public string Type { get; set; } = "text";
    public List<ProductInfo>? Data { get; set; }
}

public class ChatRequestModel
{
    public string Role { get; set; } = "user";
    public List<ChatContentItem> Contents { get; set; } = new();
}

public class ChatContentItem
{
    public string Text { get; set; } = string.Empty;
}

public enum ChatIntent
{
    ProductRequest = 0,

    Converse = 1
}
